Marlin/src/HAL/platforms.h

```diff
@@ -15,12 +15,12 @@
   #define HAL_PLATFORM HAL_LPC1768
 #elif defined(__STM32F1__) || defined(TARGET_STM32F1)
   #define HAL_PLATFORM HAL_STM32F1
-#elif defined(STM32F4) || defined(STM32F4xx)
+#elif defined(STM32F4) && !defined(STM32F4xx)
   #define HAL_PLATFORM HAL_STM32F4
 #elif defined(STM32F7) && !defined(STM32F7xx)
   #define HAL_PLATFORM HAL_STM32F7
-#elif defined(STM32F7xx)
-  #define HAL_PLATFORM HAL_STM32F7xx
+#elif defined(STM32F0xx) || defined(STM32F1xx) || defined(STM32F4xx) || defined(STM32F7xx)
+  #define HAL_PLATFORM HAL_STM32
```

Comment:
If this segment is moved up then the ! clauses above can be removed.

Reply:
Fixed. Btw: We could use the "ARDUINO_ARCH_STM32" symbol to further simplify things. But the STM32GENERIC core also defines that symbol and maybe this could lead to problems when debugging issues later on.